src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -4394,6 +4401,13 @@ module TcDeclarations =
                 members |> List.tryPick (function 
                     | SynMemberDefn.ImplicitCtor (ctorArgs = SynSimplePats.SimplePats _ as spats) -> Some spats
                     | _ -> None)
+                                
+            // Raise a new error if we try to have an interface implementation with auto properties on constructor-less types,
+            for mem in implementedAutoProperties do
+                match mem with
+                | SynMemberDefn.AutoProperty( isStatic = false; range = m) when implicitCtorSynPats.IsNone ->
+                    errorR(Error(FSComp.SR.tcAutoPropertyRequiresImplicitConstructionSequence(), m))
+                | _ -> ()
```

Comment:
I wonder if `implicitCtorSynPats.IsNone` check should wrap the loop, so we don't enumerate properties every time?

Reply:
Not just the loop, the whole constructions of `implementedAutoProperties` can be avoided (= allocations saved)

Reply:
Moved the check to `CheckMembersForm` where it already check for implicit constructor so this we can use the auto property  check in a more easy way